Python/Product/Common/Infrastructure/ProcessOutput.cs

```diff
@@ -253,7 +253,7 @@ public static ProcessOutput Run(
                 psi.Arguments = string.Join(" ",
                     arguments.Where(a => a != null).Select(QuoteSingleArgument));
             } else {
-                psi.Arguments = string.Join(" ", arguments.Where(a => a != null));
+                psi.Arguments = string.Join(" ", arguments.Where(a => a != null)); // CodeQL [SM00406] Code ql is complaining, but this is a utility class that is called with many different file names.
```

Comment:
complaining that we dont verify arguments?

Reply:
Yeah so the actual violation is https://liquid.microsoft.com/Web/Object/Read/ScanningToolWarnings/Requirements/CodeQL.SM00406#Zguide. The problem is that we can't possibly validate arguments here, it's a generic helper class to run a command line, we just join the arguments and run. Just red tape.

Reply:
so alternatively we look at all the usage points to find a set of strings.. but there will still be some dynamic ones. maybe arg templates safer.